editor/plugins/tile_set_editor_plugin.cpp

```diff
@@ -126,7 +126,7 @@ void TileSetEditor::_import_node(Node *p_node, Ref<TileSet> p_library) {
 				Transform2D shape_transform = sb->shape_owner_get_transform(E->get());
 				bool one_way = sb->is_shape_owner_one_way_collision_enabled(E->get());
 
-				shape_transform.set_origin(shape_transform.get_origin() - phys_offset);
```

Comment:
you can do:
```
 shape_transform[2] -=
```
instead if set/get origin. I still have the feeling something like this probably should use more matrix code.

Reply:
will look into it.

Reply:
Made some changes. Is it better.

Reply:
Fixed all the mistakes and problems I am very sorry for any problems. 😳